guava/src/com/google/common/base/Ascii.java

```diff
@@ -629,4 +630,78 @@ private static int getAlphaIndex(char c) {
     // Fold upper-case ASCII to lower-case and make zero-indexed and unsigned (by casting to char).
     return (char) ((c | 0x20) - 'a');
   }
+
+  /**
+   * Returns the index within {@code sequence} of the first occurrence of the
+   * specified {@code subSequence}, ignoring the case of any ASCII alphabetic characters
+   * between {@code 'a'} and {@code 'z'} or {@code 'A'} and {@code 'Z'} inclusive, or
+   * {@code -1}, if no occurence is found.
+   *
+   * @since 16.0 // TODO: which number?
+   */
+  public static int indexOfIgnoreCase(CharSequence sequence, CharSequence subSequence) {
+    return toLowerCase(sequence).indexOf(toLowerCase(subSequence));
```

Comment:
I fear this implementation may be too slow: If the length of `sequence` is `n`, the length of `subSequence` is `m`, then the time complexity is always O(n+m) or higher, because all the chars need to be converted into a string first; if we can change a letter to lower case one by one, then the execution time may be faster.

Reply:
Good point. I didn't initially think of this, but I'll definitely have a look into it.

Reply:
Just fixed that one by implementing the toLowerCase() call in the for-loop itself